src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysis.java

```diff
@@ -172,11 +169,13 @@ SecurityAnalysisResult runSync(SecurityAnalysisParameters securityAnalysisParame
         String tmpVariantId = "olf-tmp-" + UUID.randomUUID().toString();
         network.getVariantManager().cloneVariant(network.getVariantManager().getWorkingVariantId(), tmpVariantId);
         try {
-            for (Switch sw : network.getSwitches()) {
-                sw.setRetained(false);
-            }
-            for (Switch sw : allSwitchesToOpen) {
-                sw.setRetained(true);
+            if (network.getVoltageLevelStream().anyMatch(vl -> vl.getTopologyKind() == TopologyKind.NODE_BREAKER)) {
```

Comment:
So it means that we don't support hybrid networks with node/breaker and bus/breaker voltage levels?

Reply:
Not an issue but in that case I will check it at startup and throws an exception in such a case, because here I feel like a propagation between mix of node/breaker and bus/breaker would not work.

Reply:
I added support of hybrid networks